services/brig/src/Brig/Provider/API.hs

```diff
@@ -359,7 +359,9 @@ newAccount new = do
       (Code.Retries 3)
       (Code.Timeout (3600 * 24)) -- 24h
       (Just (toUUID pid))
-  wrapClientE $ Code.insert code
+  ttl <- setCodeGenerationDelaySecs <$> view settings
+  mRetryAfter <- wrapClientE $ Code.insert code ttl
+  mapM_ (throwE . verificationCodeThrottledError . VerificationCodeThrottled) mRetryAfter
```

Comment:
this would probably be a good candidate for a helper function somewhere that replaces the left side with a similarly long line.